addons/l10n_ro_efactura/models/account_move.py

```diff
@@ -0,0 +1,209 @@
+import requests
+
+from odoo import models, fields, _, api, modules, tools
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_ro_edi_document_ids = fields.One2many(
+        comodel_name='l10n_ro_edi.document',
+        inverse_name='invoice_id',
+    )
+    l10n_ro_edi_state = fields.Selection(
+        selection=[
+            ('invoice_sending', 'Sending'),
+            ('invoice_sent', 'Sent'),
+        ],
+        string='E-Factura Status',
+        compute='_compute_l10n_ro_edi_state',
+        store=True,
+    )
+    l10n_ro_edi_attachment_id = fields.Many2one(comodel_name='ir.attachment')
+
+    ################################################################################
+    # Compute Methods
+    ################################################################################
+
+    @api.depends('l10n_ro_edi_document_ids')
+    def _compute_l10n_ro_edi_state(self):
+        self.l10n_ro_edi_state = False
+        for move in self:
+            for document in move.l10n_ro_edi_document_ids.sorted():
+                if document.state in ('invoice_sending', 'invoice_sent'):
+                    move.l10n_ro_edi_state = document.state
+                    break
+
+    @api.depends('l10n_ro_edi_state')
+    def _compute_show_reset_to_draft_button(self):
+        """ Prevent user to reset move to draft when there's an
+            active sending document or a successful response has been received """
+        # EXTENDS 'account'
+        super()._compute_show_reset_to_draft_button()
+        for move in self:
+            if move.l10n_ro_edi_state in ('invoice_sending', 'invoice_sent'):
+                move.show_reset_to_draft_button = False
+
+    ################################################################################
+    # Romanian Document Shorthands & Helpers
+    ################################################################################
+
+    def _l10n_ro_edi_create_document_attachment(self, res_id, attachment_raw):
+        return self.env['ir.attachment'].sudo().create({
+            'name': self._l10n_ro_edi_get_attachment_file_name(),
+            'res_model': self._name,
+            'res_id': res_id,
+            'raw': attachment_raw,
+            'type': 'binary',
+            'mimetype': 'application/xml',
+        })
+
+    def _l10n_ro_edi_create_document_invoice_sending(self, key_loading, attachment_raw):
+        """ Shorthand for creating a `l10n_ro_edi.document` of state `invoice_sending` """
+        self.ensure_one()
+        document = self.env['l10n_ro_edi.document'].create({
+            'invoice_id': self.id,
+            'state': 'invoice_sending',
+            'key_loading': key_loading,
+        })
+        document.attachment_id = self._l10n_ro_edi_create_document_attachment(document.id, attachment_raw)
+        return document
```

Comment:
Wrong combination res_model/res_id:
res_model comes from the account.move but res_id is the one of the document.
```suggestion
    def _l10n_ro_edi_create_document_attachment(self, raw, **kwargs):
        return self.env['ir.attachment'].sudo().create({
            'name': self._l10n_ro_edi_get_attachment_file_name(),
            'res_model': self._name,
            'res_id': self.id,
            'raw': raw,
            'type': 'binary',
            'mimetype': 'application/xml',
            **kwargs,
        })

    def _l10n_ro_edi_create_document_invoice_sending(self, key_loading, attachment_raw):
        """ Shorthand for creating a `l10n_ro_edi.document` of state `invoice_sending` """
        self.ensure_one()
        document = self.env['l10n_ro_edi.document'].create({
            'invoice_id': self.id,
            'state': 'invoice_sending',
            'key_loading': key_loading,
        })
        document.attachment_id = self._l10n_ro_edi_create_document_attachment(
            raw,
            res_model=document._name,
            res_id=document.res_id,
        )
        return document
```